Game/src/Enemy.java

```diff
@@ -0,0 +1,172 @@
+                flag = true;
+            }
+            vX += flag ? speed : -speed;
+
+        } else if (logicMode == LogicMode.FOLLOW) {
+            if (getX() < MainApplication.player.getX()) {
+                vX += speed;
+            } else if (getX() > MainApplication.player.getX()) {
+                vX -= speed;
+            }
+        } else if (logicMode == LogicMode.PATROL_AB){
+
+            if (getX() + patrolDistance > MainApplication.player.getX() && getX() - patrolDistance < MainApplication.player.getX()) {
+                patrolTriggered = true;
+            }
+
+            if (!patrolTriggered) {
+                if (getX() >= pointB) {
+                    flag = false;
+                } else if (getX() <= pointA) {
+                    flag = true;
+                }
+                vX += flag ? speed : -speed;
+            } else {
+                if (getX() < MainApplication.player.getX()) {
+                    vX += speed;
+                } else if (getX() > MainApplication.player.getX()) {
+                    vX -= speed;
+                }
+            }
+
+        }
+
+        setX(getX() + vX);
+        setY(getY() + vY);
+
+        hpBar.setX(getX());
+        hpBar.setY(getY() - 8);
+
+    }
+
+    public boolean checkCollision(Player player){
+
+        boolean collided = false;
+
+        if (player.getX() + player.getFitWidth() >= getX() && player.getX() < getX() + getWidth()
+                && player.getY() + player.getFitHeight() > getY() && player.getY() < getY() + getHeight()) {
+            collided = true;
+        }
+
+        if (player.getX() <= getX() + getWidth() && player.getX() + player.getFitWidth() > getX()
+                && player.getY() + player.getFitHeight() > getY() && player.getY() < getY() + getHeight()) {
+            collided = true;
+        }
+
+        if (player.getY() <= getY() + getHeight() && player.getY() + player.getFitHeight() > getY()
+                && player.getX() + player.getFitWidth() > getX() && player.getX() < getX() + getWidth()) {
+            collided = true;
+        }
+
+        if (player.getY() + player.getFitHeight() >= getY() && player.getY() < getY() + getHeight()
+                && player.getX() + player.getFitWidth() > getX() && player.getX() < getX() + getWidth()) {
+            collided = true;
+        }
+
+        return collided;
+
+    }
+
+    public double getWidth(){
+        return getFitWidth();
+    }
+
+    public double getHeight(){
+        return getFitHeight();
+    }
+
+    public LogicMode getLogicMode(){
+        return logicMode;
+    }
+
+    public void setLogicMode(LogicMode logicMode){
+        this.logicMode = logicMode;
+    }
+
+    public void setPointAB(double a, double b){
+        this.pointA = a;
+        this.pointB = b;
+        this.oPointA = a;
+        this.oPointB = b;
+    }
+
+    public void setPatrolDistance(double distance){
+        this.patrolDistance = distance;
+    }
+
+}
```

Comment:
Add a new line otherwise it could be incompatible with older editors/compilers. Just good practice